services/license-store.js

```diff
@@ -0,0 +1,99 @@
+const assert = require('assert');
+
+class License {
+  constructor(id, name, groups, compatibility, regexp, url) {
+    assert(typeof id === 'string' && id.length > 0, 'License: Invalid "id" provided');
+    assert(typeof name === 'string' && name.length > 0, 'License: Invalid "name" provided');
+    assert(Array.isArray(groups), 'License: Invalid "groups" provided');
+    assert(Array.isArray(compatibility), 'License: Invalid "compatibility" provided');
+    assert(regexp instanceof RegExp, 'License: Invalid "regexp" provided');
+    assert(!url || typeof url === 'string', 'License: Invalid "url" provided');
```

Comment:
I like the way of validating the input data, I'd just propose to extract this into a separate function!